configdialog.cpp

```diff
@@ -518,6 +532,23 @@ void PaletteCustomizeDialog::get_and_save_color(int palette_index) {
 	}
 }
 
+std::wstring PaletteCustomizeDialog::export_palettes() {
+	auto all_palettes = m_palette_repo.get_all_custom_palettes();
+	std::wstring palettes_string = L"";
+
+	for (const auto &palette : all_palettes) {
+		auto name = palette.name;
+		auto *data = palette.data;
+
+		palettes_string.append(palette.name + L": ");
+		palettes_string.append(m_palette_repo.serialize(*data));
+		palettes_string.append(L"\r\n");
```

Comment:
Line breaks are a text encoding nightmare and it makes me very anxious to put in something that will be pasted around and sent over the internet. The clipboard could fuck it up, someone could paste it into an editor that replaces \r\n with \n, someone could paste it into discord and that replaces \r\n with \n, it's just bad. I suggest we pick a different separator.

Now this of course means your export code is all one line, which is a lot uglier, but that might actually be a good thing. It's nice for a data transfer code to be indecipherable nonsense so that people treat it like an opaque box of horrors they don't want to touch. If people look at and they think they know how it works, they might be empowered to touch it or reformat it so it looks nicer and then whoops they fucked it up.

(To that end you could also get rid of the space after the palette name, since I think all it's doing is just making it look nicer)

Reply:
They pretty much are there _just_ to make it look nicer. I don't intend on making them required.

Also, windows did not appreciate this being just "\n".